hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/operations/HederaEvmOperationsUtilV038.java

```diff
@@ -53,15 +55,22 @@ static Operation.OperationResult addressCheckExecution(
             @NonNull final Supplier<Operation.OperationResult> supplierExecution,
             @NonNull final BiPredicate<Address, MessageFrame> addressValidator,
             @NonNull final Predicate<Address> systemAccountDetector,
-            @NonNull final Supplier<Operation.OperationResult> systemAccountExecutionSupplier) {
+            @NonNull final Supplier<Operation.OperationResult> systemAccountExecutionSupplier,
+            @NonNull final EvmProperties evmProperties) {
         try {
             final var address = Words.toAddress(supplierAddressBytes.get());
             if (systemAccountDetector.test(address)) {
                 return systemAccountExecutionSupplier.get();
             }
-            if (Boolean.FALSE.equals(addressValidator.test(address, frame))) {
-                return new Operation.OperationResult(
-                        supplierHaltGasCost.getAsLong(), HederaExceptionalHaltReason.INVALID_SOLIDITY_ADDRESS);
+            // skip target entity existing check when
+            // evm >= 0.46 or FF is enabled or the target is grandfather contract
+            if (!evmProperties.evmVersion().equals(EVM_VERSION_0_46)
```

Comment:
This three-part condition seems duplicated in man places, would be good to encapsulate as a single method on `EvmProperties`.

Reply:
Done.